oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -32,56 +55,24 @@ public void setPosition(Vector2d position) {
 
     public void move(MoveDirection direction){
         switch (direction){
-            case RIGHT -> setOrientation(orientation.next());
-            case LEFT -> setOrientation(orientation.previous());
+            case RIGHT -> this.orientation = this.orientation.next();
+            case LEFT -> this.orientation = this.orientation.previous();
             case FORWARD -> {
-                switch (orientation){
-                    case NORTH -> {
-                        if (getPosition().add(new Vector2d(0,1)).precedes(new Vector2d(4,4))){
-                            setPosition(position.add(new Vector2d(0,1)));}
-                    }
-                    case SOUTH ->{
-                        if (getPosition().add(new Vector2d(0,-1)).follows(new Vector2d(0,0))){
-                            setPosition(position.add(new Vector2d(0,-1)));
-                        }
-                    }
-                    case EAST -> {
-                        if (getPosition().add(new Vector2d(1,0)).precedes(new Vector2d(4,4))){
-                            setPosition(position.add(new Vector2d(1,0)));
-                        }
-                    }
-                    case WEST -> {
-                        if (getPosition().add(new Vector2d(-1,0)).follows(new Vector2d(0,0))){
-                            setPosition(position.add(new Vector2d(-1,0)));
-                        }
-                    }
+                Vector2d  vector1 = this.position.add(Objects.requireNonNull(this.orientation.toUnitVector()));
+                if (map.canMoveTo(vector1) && !map.isOccupied(vector1)){
+                    this.position=vector1;
                 }
             }
             case BACKWARD -> {
-                switch (orientation){
-                    case NORTH -> {
-                        if (getPosition().add(new Vector2d(0,-1)).follows(new Vector2d(0,0))){
-                            setPosition(position.add(new Vector2d(0,-1)));
-                        }
-                    }
-                    case SOUTH -> {
-                        if (getPosition().add(new Vector2d(0,1)).precedes(new Vector2d(4,4))){
-                            setPosition(position.add(new Vector2d(0,1)));}
-                    }
-                    case EAST -> {
-                        if (getPosition().add(new Vector2d(-1,0)).follows(new Vector2d(0,0))){
-                            setPosition(position.add(new Vector2d(-1,0)));
-                        }
-                    }
-                    case WEST -> {
-                        if (getPosition().add(new Vector2d(1,0)).precedes(new Vector2d(4,4))){
-                            setPosition(position.add(new Vector2d(1,0)));
-                        }
-                    }
+                Vector2d  vector1 = this.position.add(Objects.requireNonNull(this.orientation.toUnitVector()).opposite());
+                if (map.canMoveTo(vector1) && !map.isOccupied(vector1)){
```

Comment:
można by tak zaimplementować `canMoveTo()` żeby również sprawdzało `isOccupied()`, tak na logikę to ma sens - nie mogę tam ruszyć, jeśli pozycja jest zajęta.